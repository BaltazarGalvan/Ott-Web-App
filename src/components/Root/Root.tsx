import React, { FC } from 'react';
import { Route, Switch } from 'react-router-dom';

import Home from '../../screens/Home/Home';

// Mock screens

const PlaylistScreen = () => {
  return (
    <>
      <span>PlaylistScreen</span>
    </>
  );
};
const Settings = () => {
  return (
    <>
      <span>Settings</span>{' '}
    </>
  );
};

// Mock screens

type Props = {
  error?: Error | null;
};

const Root: FC<Props> = ({ error }: Props) => {
  if (error) {
    return <div>Error!</div>;
  }

  return (
    <Switch>
      <Route path="/" component={Home} exact />
      <Route path="/p/:id" component={PlaylistScreen} exact />
      <Route path="/u" component={Settings} exact />
    </Switch>
  );
};

export default Root;
