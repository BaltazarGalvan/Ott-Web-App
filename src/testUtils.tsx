import { BrowserRouter as Router } from 'react-router-dom';
import React, { ReactNode, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';

interface WrapperProps {
  children?: ReactNode;
}

export const wrapper = ({ children }: WrapperProps) => <Router>{children}</Router>;

const customRender = (ui: ReactElement, options?: RenderOptions) => render(ui, { wrapper, ...options });

export const mockWindowLocation = (path: string) => {
  Object.defineProperty(window, 'location', {
    value: {
      pathname: path,
      assign: jest.fn(),
    },
  });
};

export const mockMatchMedia = () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
    })),
  });
};

export { customRender as render };
