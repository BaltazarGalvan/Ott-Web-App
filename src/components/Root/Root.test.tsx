import React from 'react';

import { mockWindowLocation, render } from '../../testUtils';

import Root from './Root';

describe('<Root />', () => {
  it('renders and matches snapshot', () => {
    mockWindowLocation('/');
    const { container } = render(<Root />);

    expect(container).toMatchSnapshot();
  });

  it('renders error page when error prop is passed', () => {
    mockWindowLocation('/');
    const error = new Error();
    const { container } = render(<Root error={error} />);

    expect(container).toMatchSnapshot();
  });
});
