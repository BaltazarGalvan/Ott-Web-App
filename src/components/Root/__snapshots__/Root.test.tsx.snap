// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Root /> renders and matches snapshot 1`] = `
<div>
  <div>
    <header
      class="header static"
    >
      <div
        class="container"
      >
        <div
          aria-label="open menu"
          class="menu"
          role="button"
        >
          <svg
            aria-hidden="true"
            class="icon"
            focusable="false"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g>
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </g>
          </svg>
        </div>
        <nav
          aria-label="menu"
          class="nav"
        >
          <a
            class="link"
            href="/"
          >
            <span>
              Home
            </span>
          </a>
          <a
            class="link"
            href="/u"
          >
            <span>
              Settings
            </span>
          </a>
        </nav>
        <div
          class="search"
        />
      </div>
    </header>
    <div
      class="backdrop"
    />
    <div
      class="sidebar"
    >
      <div
        aria-label="close menu"
        class="group"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="icon"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </g>
        </svg>
      </div>
      <nav
        class="group"
      >
        <a
          class="link"
          href="/"
        >
          <span>
            Home
          </span>
        </a>
        <hr
          class="divider"
        />
        <a
          class="link"
          href="/"
        >
          <span>
            Settings
          </span>
        </a>
      </nav>
    </div>
  </div>
</div>
`;

exports[`<Root /> renders error page when error prop is passed 1`] = `
<div>
  <div>
    Error!
  </div>
</div>
`;
